### PR TITLE
add HLS to shell env

### PR DESCRIPTION
**Pull request by yaitskov**
I started looking into megaparsec development and noticed that IDE (`code .`) inside `nix develop` doesn't have a language server.

### CODE REVIEW

**Comment by mrkkrp** [comment]
I'm a bit ambivalent about this one. I usually have HLS globally installed.

File: flake.nix [hunk 1]
```diff
@@ -141,7 +141,8 @@
         megaparsec = sdistTarball haskellPackages.megaparsec;
         megaparsec-tests = sdistTarball haskellPackages.megaparsec-tests;
       };
-
+      haskellLanguageServer = pkgs.haskell.lib.overrideCabal haskellPackages.haskell-language-server
```
Comment (mrkkrp) [inline flake.nix:144]: Why is this necessary?

Comment (yaitskov) [reply flake.nix:144]: Nixshell provides HLS, so `code .` in shell gets proper HLS for GHC.

Comment (mrkkrp) [reply flake.nix:144]: My question was rather specifically why `enableSharedExecutables = true` should be set, why not use `haskellLanguageServer` as it is?

**Comment by mrkkrp** [comment]
Closing due to lack of activity (it doesn't seem to be very important and my last question is still not answered).
